zerver/worker/queue_processors.py

```diff
@@ -1153,6 +1154,15 @@ def failure_processor(event: Dict[str, Any]) -> None:
             )
             user_profile = get_user_profile_by_id(event["user_profile_id"])
             reactivate_user_if_soft_deactivated(user_profile)
+        elif event["type"] == "scrub_deactivated_realms":
+            realms_to_scrub = Realm.objects.filter(
```

Comment:
I think we should have the event be for just _one_ `realm_id`, which is provided in the `event` -- and move this `Realm.objects.filter()` into the management command.  That "chunks" the work into smaller events.